assets/src/components/detours/detourDropdown.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react"
+import { DropdownItem, DropdownMenu } from "../map/dropdown"
+import { Route, RoutePattern, Shape } from "../../schedule"
+import { Popup } from "react-leaflet"
+import { PointTuple } from "leaflet"
+import { LatLngLiteral } from "leaflet"
+
+export interface StartDetourProps {
+  routeName: string
+  routeDescription: string
+  routeOrigin: string
+  routeDirection: string
+  shape: Shape
+  center: LatLngLiteral
+  zoom: number
+}
+
+export interface DetourDropdownProps {
+  setShouldShowPopup: (newValue: boolean) => void
+  routePatternForVehicle: RoutePattern | null
+  route: Route | null
+  onStartDetour?: (props: StartDetourProps) => void
+  onClick: (props: StartDetourProps) => void
+  center: LatLngLiteral
+  zoom: number
+}
+
+export const DetourDropdown = ({
+  routePatternForVehicle,
+  route,
+  onClick,
+  center,
+  zoom,
+}: DetourDropdownProps) => {
+  // This offset is here because, due to a limitation of Leaflet
+  // popups, we weren't able to render the popup at the bottom-right
+  // corner of the marker, where it's supposed to go. This effectively
+  // renders it centered and above the marker, and then uses the
+  // offset to reposition it to the bottom-right corner.
+  const dropdownOffset: PointTuple = [140, 97]
+
+  if (!routePatternForVehicle || !route) {
+    return <></>
+  }
+
+  const routeName = route.name
+
+  const routeDescription = routePatternForVehicle.headsign
+
+  const routeOrigin = routePatternForVehicle.name
+
+  const routeDirection =
+    route.directionNames[routePatternForVehicle.directionId]
+
+  const shape = routePatternForVehicle.shape
+
+  if (!routeDescription || !shape) {
+    return <></>
+  }
```

Comment:
suggestion: I believe you can return null here instead of constructing a document fragment
```suggestion
  if (!routePatternForVehicle || !route) {
    return null
  }

  const routeName = route.name

  const routeDescription = routePatternForVehicle.headsign

  const routeOrigin = routePatternForVehicle.name

  const routeDirection =
    route.directionNames[routePatternForVehicle.directionId]

  const shape = routePatternForVehicle.shape

  if (!routeDescription || !shape) {
    return null
  }
```